src/app/(proper_react)/redesign/(authenticated)/user/dashboard/DashboardTopBanner/DashboardTopBannerContent.tsx

```diff
@@ -0,0 +1,490 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { DashboardTopBannerProps } from ".";
+import styles from "./DashboardTopBannerContent.module.scss";
+import {
+  UserDashboardState,
+  getUserDashboardState,
+} from "../getUserDashboardState";
+import {
+  StepLink,
+  getRelevantGuidedSteps,
+} from "../../../../../../functions/server/getRelevantGuidedSteps";
+import { ProgressCard } from "../../../../../../components/client/ProgressCard";
+import { Button } from "../../../../../../components/server/Button";
+import { useL10n } from "../../../../../../hooks/l10n";
+import PremiumButton from "../../../../../../components/client/PremiumButton";
+
+export interface ContentProps {
+  relevantGuidedStep: StepLink;
+  hasExposures: boolean;
+  hasUnresolvedBreaches: boolean;
+  hasUnresolvedBrokers: boolean;
+  isEligibleForFreeScan: boolean;
+  isEligibleForPremium: boolean;
+  isPremiumUser: boolean;
+  scanInProgress: boolean;
+  onShowFixed: () => void;
+}
+
+export const DashboardTopBannerContent = (props: DashboardTopBannerProps) => {
+  const l10n = useL10n();
+
+  const {
+    tabType,
+    bannerData,
+    stepDeterminationData,
+    hasExposures,
+    hasUnresolvedBreaches,
+    hasUnresolvedBrokers,
+    isEligibleForFreeScan,
+    isEligibleForPremium,
+    isPremiumUser,
+    scanInProgress,
+    onShowFixed,
+  } = props;
+
+  if (tabType === "fixed") {
+    return (
+      <ProgressCard
+        resolvedByYou={bannerData.dataBreachFixedExposuresNum}
+        autoRemoved={bannerData.dataBrokerFixedNum}
+        totalNumExposures={bannerData.totalExposures}
+      />
+    );
+  }
+
+  const relevantGuidedStep = getRelevantGuidedSteps(
+    stepDeterminationData
+  ).current;
```

Comment:
Heads-up depending on whose PR gets merged first: in https://github.com/mozilla/blurts-server/pull/3485 I removed `.current` and renamed this to `getNextGuidedStep`.

Reply:
Oh, actually, maybe that's just in a PR I haven't even submitted yet. Then you'll probably be first :grin: